app/src/test/java/org/greenstand/android/TreeTracker/managers/LocationDataCapturerTest.kt

```diff
@@ -56,14 +62,83 @@ class LocationDataCapturerTest {
 
         locationDataCapturer.turnOnTreeCaptureMode()
 
-        assertNotNull(locationDataCapturer.generatedTreeUuid)
+        assertNotNull("Tree UUID is generated", locationDataCapturer.generatedTreeUuid)
     }
 
     @Test
     fun turnOffTreeCaptureMode() {
 
         locationDataCapturer.turnOffTreeCaptureMode()
 
-        assertNull(locationDataCapturer.generatedTreeUuid)
+        assertNull("Convergence is reset to null", locationDataCapturer.convergence)
+        assertNull("Tree UUID is reset to null", locationDataCapturer.generatedTreeUuid)
+    }
+
+    @Test
+    fun noConvergenceUntilMinLocationSize() {
+
+        val locationsLiveData = MutableLiveData<Location>()
+        every { locationUpdateManager.locationUpdateLiveData } returns locationsLiveData
+
+        locationDataCapturer.start()
+        locationDataCapturer.turnOnTreeCaptureMode()
+
+        // Assuming a minimum of 5 Location data required for evaulating
+        // convergence
+        for (i in 1..4) {
+            val location = mockk<Location>(relaxed = true)
+            locationsLiveData.postValue(location)
+        }
+
+        assertFalse(locationDataCapturer.convergenceWithinRange)
+    }
+
+    val locationValuesLowVariance = listOf(
+        Pair(-122.08400001, 37.42149486),
+        Pair(-122.08400111, 37.42149487),
+        Pair(-122.08400021519, 37.42149483),
+        Pair(-122.08400021999, 37.42149489),
+        Pair(-122.0840086753, 37.42149481)

```

Comment:
This is fine, but it can also be done like:
`listOf(
123 to 456, 
678 to 901
)`

Reply:
I don't understand what you are pointing to here. You want to use listOf instead of Pair?

Reply:
I'm saying instead of
`Pair(a, b)`
you can do
`a to b`

Reply:
Just wanted a simple pair  of longitude, latitude values. Map would make the intent unclear.